src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -804,20 +802,26 @@ DeferredErrorIfUnsupportedRecurringTuplesJoin(
 				continue;
 			}
 
-			if (ShouldRecurseForRecurringTuplesJoinChecks(outerrel) &&
-				RelationInfoContainsRecurringTuples(plannerInfo, outerrel, &recurType))
+			if (RelationInfoContainsOnlyRecurringTuples(plannerInfo, outerrel))
```

Comment:
This could be an expensive call.

Reply:
Running all the regression tests with measuring the time spend in `DeferredErrorIfUnsupportedRecurringTuplesJoin ()`, none of the queries took more than 1 msecs (I think measuring less than 1 msec is not so easy with PG functions?).

Even if I have 250 tables and for all the loops inside `DeferredErrorIfUnsupportedRecurringTuplesJoin()` go until the end , the total planning time:  53 seconds and  `DeferredErrorIfUnsupportedRecurringTuplesJoin`: 9msec

So, overall it doesn't seem to be a bottleneck